pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -169,3 +147,305 @@ func TestDeletePods(t *testing.T) {
+				Annotations: map[string]string{kubetypes.ConfigHashAnnotationKey: "test"},
+			},
+		},
+		mirrorPod: &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:         types.UID("987654321"),
+				Name:        "baz",
+				Namespace:   "default",
+				Annotations: map[string]string{kubetypes.ConfigMirrorAnnotationKey: "test"},
+			},
+		},
+		expected: false,
+	}}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			actual := podManager.IsMirrorPodOf(test.mirrorPod, test.pod)
+
+			if actual != test.expected {
+				t.Errorf("actual: %t, expected: %t", actual, test.expected)
+			}
+		})
+	}
+}
+
+func TestGetMirrorPodByPod(t *testing.T) {
+	testcases := []struct {
+		name              string
+		pod               *v1.Pod
+		expectedMirrorPod *v1.Pod
+		expectedOk        bool
+	}{{
+		name:              "test pod found",
+		pod:               newPod(types.UID("987654321"), "bar", "default", Static),
+		expectedMirrorPod: newPod(types.UID("123456789"), "bar", "default", Mirror),
+		expectedOk:        true,
+	}, {
+		name:              "test pod not found",
+		pod:               newPod(types.UID("987654321"), "bar", "default", Static),
+		expectedMirrorPod: newPod(types.UID("123456789"), "baz", "default", Mirror),
+		expectedOk:        false,
+	}}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.AddPod(test.expectedMirrorPod)
+			actualMirrorPod, actualOk := podManager.GetMirrorPodByPod(test.pod)
+
+			if actualOk != test.expectedOk {
+				t.Errorf("actualOk: %t, expectedOk: %t", actualOk, test.expectedOk)
+			}
+
+			if actualOk && !reflect.DeepEqual(actualMirrorPod, test.expectedMirrorPod) {
+				t.Errorf("actualMirrorPod: %#v, expectedMirrorPod: %#v", actualMirrorPod, test.expectedMirrorPod)
+			}
+		})
+	}
+}
+
+func TestGetPodByMirrorPod(t *testing.T) {
+	testcases := []struct {
+		name        string
+		mirrorPod   *v1.Pod
+		expectedPod *v1.Pod
+		expectedOk  bool
+	}{{
+		name:        "test pod found",
+		mirrorPod:   newPod(types.UID("987654321"), "bar", "default", Mirror),
+		expectedPod: newPod(types.UID("123456789"), "bar", "default", Static),
+		expectedOk:  true,
+	}, {
+		name:        "test pod not found",
+		mirrorPod:   newPod(types.UID("987654321"), "bar", "default", Mirror),
+		expectedPod: newPod(types.UID("123456789"), "baz", "default", Static),
+		expectedOk:  false,
+	}}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.AddPod(test.expectedPod)
+			actualPod, actualOk := podManager.GetPodByMirrorPod(test.mirrorPod)
+
+			if actualOk != test.expectedOk {
+				t.Errorf("actualOk: %t, expectedOk: %t", actualOk, test.expectedOk)
+			}
+
+			if actualOk && !reflect.DeepEqual(actualPod, test.expectedPod) {
+				t.Errorf("actualMirrorPod: %#v, expectedMirrorPod: %#v", actualPod, test.expectedPod)
```

Comment:
Should we use the `actualPod` instead of the `actualMirrorPod`, as the value is not the mirrorPod one ?